### DISCUSSION TITLE
fail to search on all shard leaders, err=All attempts results:\nattempt #1:code: NotReadyServe, error: err: rpc error: code = Canceled desc = context canceled\n

### DISCUSSION THREAD

**Discussion by ReganWz** [milvus-io/milvus#26814] [Q&A and General discussion]
03 01:43:50.585 +00:00] [WARN] [proxy/task_scheduler.go:473] ["Failed to execute task: "] [error="fail to search on all shard leaders, err=All attempts results:\nattempt #1:code: NotReadyServe, error: err: rpc error: code = Canceled desc = context canceled\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:319 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call\n/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/client/client.go:254 github.com/milvus-io/milvus/internal/distributed/querynode/client.(*Client).Search\n/go/src/github.com/milvus-io/milvus/internal/proxy/task_search.go:567 github.com/milvus-io/milvus/internal/proxy.(*searchTask).searchShard\n/go/src/github.com/milvus-io/milvus/internal/proxy/task_policies.go:132 github.com/milvus-io/milvus/internal/proxy.mergeRoundRobinPolicy.func1\n/usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit\n\nattempt #2:context canceled\n"] [traceID=88260c6ef4249b5]

**Comment by xiaofan-luan** [comment]
please offer detailed logs of querynode, querycoord and proxy.
current log is not sufficient for further investigation

**Reply by ReganWz** [reply]
this happened in 2023.09-02
[podlog.zip](https://github.com/milvus-io/milvus/files/12505426/podlog.zip)

**Comment by xiaofan-luan** [comment]
the log you offered is not completed.
from the proxy, we do see some slow queries form 09/02
but  for querynode all the logs are 09/03.

could you issue some query request and check?

**Comment by xiaofan-luan** [comment]
I've seen that the search takes very long time(10s)
this might be due to two reasons
1. your rocksmq seems to be very slow, what disk type you are using ?
2. it seems there are many partitions in your collection, how many partitions do you have in total and how many data are there in each partition?

**Reply by ReganWz** [reply]
1.Mechanical disk
2.There are only 100 partitions, each with 1000 data
